homeassistant/components/emoncms/sensor.py

```diff
@@ -91,12 +89,12 @@ def setup_platform(
     exclude_feeds = config.get(CONF_EXCLUDE_FEEDID)
     include_only_feeds = config.get(CONF_ONLY_INCLUDE_FEEDID)
     sensor_names = config.get(CONF_SENSOR_NAMES)
-    interval = config.get(CONF_SCAN_INTERVAL)
 
     if value_template is not None:
         value_template.hass = hass
 
-    data = EmonCmsData(hass, url, apikey, interval)
+    assert url is not None and apikey is not None  # Keep mypy happy
```

Comment:
If they are compulsory, then it should be using `config[key]` not `config.get(key)`

Reply:
That's a really good point. I had forgotten I was looking at a dict, rather than some opaque data class :laughing: They certainly are compulsory, and marked as `Required` in the schema.

Reply:
I'd be also worried about this PR getting bigger and bigger.
- Adding type hints can usually be a single PR.
- Adding `unique_id` probably can be also it's own PR
- and maybe the `state_class` in a separate PR.

Reply:
OK, if that is the convention. It seems like there will be a lot of admin overheard for adding one line of code at a time but if that's the standard I am happy to comply.

I don't have the time to pull it all apart this week though. Maybe next week.

Reply:
It's not compulsory - but it makes reviewing ssssoooo much easier if the PR is smaller.
It also means that if there is a difference of opinion on one point then the other points can be merged anyway.

Reply:
OK yes I see your point. I guess my view was that a lot of this is just tidying things up and doesn't really improve the user experience, so it makes sense to carry it along with features rather than bothering reviewers to look through multiple PRs.

I am certainly expecting to move the integration to use a config flow in a separate PR.

Reply:
Actually if you look at the instructions when making a new PR:
```
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
```
Which ended up with:
![image](https://user-images.githubusercontent.com/1858925/215889961-0165cba3-d27a-4ceb-9d9c-5b693012873a.png)

Don't get me wrong, I can't say I meet these instructions, many times dependency bump is also a bugfix, etc...

However, code quality are usually easy to split out to a new PR, they are also good candidates for people to review since unlike bug fixes which change the code and understanding for the change, code quality usually try to keep the same functionality and is easier to review. 

There is no overhead from creating multiple PRs (@epenet  is expert in creating huge number of PRs ❤️), the opposite, it makes easier to review small PRs, although this PR is still in a reasonable size.

Reply:
OK, that makes sense, thanks for explaining :smile:  Unfortunately I have lots of work to do this week but I hope I will be able to sort out my mess next week :smile:  Thanks again! :heart: